crates/sui-core/src/epoch/randomness.rs

```diff
@@ -278,13 +278,15 @@ impl RandomnessManager {
             "random beacon: starting from next_randomness_round={}",
             rm.next_randomness_round.0
         );
-        for result in tables.randomness_rounds_pending.safe_iter() {
-            let (round, _) = result.expect("typed_store should not fail");
+        if highest_completed_round + 1 < rm.next_randomness_round {
             info!(
-                "random beacon: resuming generation for randomness round {}",
-                round.0
+                "random beacon: resuming generation for randomness rounds from {} to {}",
+                highest_completed_round + 1,
+                rm.next_randomness_round - 1,
             );
-            network_handle.send_partial_signatures(committee.epoch(), round);
+            for r in highest_completed_round.0 + 1..rm.next_randomness_round.0 {
```

Comment:
if the gap is large, this can flood the network, no?

Reply:
No because of the concurrency limit in RandomnessNetworkLoop